src/coffee/sync/utils/product.coffee

```diff
@@ -248,15 +247,15 @@ class ProductUtils extends BaseUtils
         index = key.substring(1)
       if index
         delete price.discounted # we don't need this for mapping the action
-
         # if id is unchanged, we initiate a changePrice action with priceId
         if not _.isArray(price) and not price.id and old_variant.prices.length > index
-          changeAction = @_buildChangePriceActionWithPriceId(new_variant, index)
+          changeAction = @_buildChangePriceActionWithPriceId(old_variant, new_variant, index)
           actions.push changeAction if changeAction
         else
           delete price.id # delete the id so it doesn't conflict with other actions
-          if _.size(price) is 1 and _.size(price.value) is 1 and _.has(price.value, 'centAmount')
-            changeAction = @_buildChangePriceAction(price.value.centAmount, old_variant, index)
+          # build changePrice update action with price selection
+          if not _.isArray(price) and _.size(price) and not _.has(price, 'country') and not _.has(price, 'channel') and not _.has(price, 'customerGroup') and not _.isArray(price.value?.currencyCode) and old_variant.prices.length > index
```

Comment:
Can this check be formatted/indented so that it is not in single line and thus better to read? Some comments per check would be nice also as it is not obvious what particular check do and why. For example:
- Why price would be not an array and simultaneously have a array size greater than 0? `_.isArray(price) and _.size(price)`?
- Same question can be applied for `currencyCode`

Why we need two `if` statements in line 251 and 257? Would not simple `old_variant.prices.length > index` be enough for both cases?

Reply:
> Why price would be not an array and simultaneously have a array size greater than 0? _.isArray(price) and _.size(price)

Checking that it isn't an empty `object`, but yeah, I will add comments to the code